rllib/algorithms/algorithm_config.py

```diff
@@ -901,17 +910,23 @@ def validate(self) -> None:
                 # compatibility for now. User only needs to set num_rollout_workers.
                 self.input_config["parallelism"] = self.num_rollout_workers or 1
 
-        # resolve rl_module_spec class
-        if self._enable_rl_module_api and self.rl_module_spec is None:
-            self.rl_module_spec = self.get_default_rl_module_spec()
-            if not isinstance(
-                self.rl_module_spec, (SingleAgentRLModuleSpec, MultiAgentRLModuleSpec)
-            ):
-                raise ValueError(
-                    "rl_module_spec must be an instance of "
-                    "SingleAgentRLModuleSpec or MultiAgentRLModuleSpec."
-                    f"Got {type(self.rl_module_spec)} instead."
-                )
+        if self._enable_rl_module_api:
+            if self.rl_module_spec is not None:
+                # Merge provided RL Module spec class with defaults
+                _check_rl_module_spec(self.rl_module_spec)
+                default_rl_module_spec = self.get_default_rl_module_spec()
+                # We can only merge if we have SingleAgentRLModuleSpecs.
+                # TODO(Artur): Support merging for MultiAgentRLModuleSpecs.
+                if (
+                    type(default_rl_module_spec) is SingleAgentRLModuleSpec
+                    and type(self.rl_module_spec) is SingleAgentRLModuleSpec
+                ):
+                    _check_rl_module_spec(default_rl_module_spec)
```

Comment:
We should just check the default_rl_module_spec right off the bat. That's the first thing we should check if enbale_api is enabled.

Reply:
so that you don't have to do this here.